code/graph_algorithms/src/hamiltonian_cycle/hamiltonian_cycles.cpp

```diff
@@ -0,0 +1,130 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <math.h>
+
+using namespace std;
+
+/**
+    prints all the hamiltonian paths and cycles of the graph
+**/
+void printPathsAndCycles(vector<vector<string> > mN,vector<vector<string> > m1, int n)
+{
+    cout <<"\nThe hamiltonian paths of the graph are : ";
+    vector<string> paths;
+    vector<string> edges;
+    for (int i=0; i<n; i++)
+    {
+        for (int j=0; j<n; j++)
+        {
+            if (mN[i][j] != "0")            // prints the paths from the last mN matrix
+            {
+                cout <<  mN[i][j] <<", " ;
+                paths.push_back(mN[i][j]);
+            }
+
+            if (m1[i][j] != "0")             // stores the edges of the m1 matrix
+            {
+                edges.push_back(m1[i][j]);
+            }
+
+        }
+    }
+    cout <<"\n";
+    string edge1;
+    string edge2;
+    cout <<"\nThe hamiltonian cycles of the graph are : ";
+    for (int i=0; i<paths.size(); i++)
+    {
+        edge2 = paths[i].substr(paths[i].length()-1,1)+paths[i].substr(0,1); // edge that consists of the 1st and the last path nodes
```

Comment:
same as above , spaces between operators and operands. Add space after comma as well.